graphql/mutation_resolver.go

```diff
@@ -175,37 +174,32 @@ func (r *mutationResolver) CreateDistro(ctx context.Context, opts CreateDistroIn
 	}, nil
 }
 
-// SaveDistroSection is the resolver for the saveDistroSection field.
+// SaveDistroSection is the resolver for the saveDistroSection field. The entire distro object is provided as input (not just the updated fields) in order to validate all distro settings.
 func (r *mutationResolver) SaveDistroSection(ctx context.Context, opts SaveDistroInput) (*SaveDistroSectionPayload, error) {
 	usr := mustHaveUser(ctx)
-	distroChanges := opts.Changes.ToService()
+	d := opts.Distro.ToService()
 
-	originalDistro, err := distro.FindOneId(ctx, opts.DistroID)
+	settings, err := evergreen.GetConfig(ctx)
+	validationErrs, err := validator.CheckDistro(ctx, d, settings, false)
 	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("finding distro '%s': %s", opts.DistroID, err.Error()))
-	}
-	if originalDistro == nil {
-		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("unable to find distro '%s'", opts.DistroID))
+		return nil, InternalServerError.Send(ctx, err.Error())
 	}
-
-	if err = validator.ValidateDistroSection(ctx, originalDistro, distroChanges, opts.Section); err != nil {
-		return nil, InputValidationError.Send(ctx, fmt.Sprintf("validating changes for distro '%s': %s", opts.DistroID, err.Error()))
+	if len(validationErrs) != 0 {
+		return nil, InputValidationError.Send(ctx, fmt.Sprintf("validating changes for distro '%s': '%s'", d.Id, validationErrs.String()))
 	}
 
-	updatedDistro, err := distro.UpdateDistroSection(ctx, originalDistro, distroChanges, opts.Section, usr.Username())
-	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("updating distro '%s': %s", opts.DistroID, err.Error()))
+	if err = d.ReplaceOne(ctx); err != nil {
+		return nil, InternalServerError.Send(ctx, err.Error())
 	}
-	apiDistro := &restModel.APIDistro{}
-	apiDistro.BuildFromService(*updatedDistro)
+	event.LogDistroModified(d.Id, usr.Username(), d.NewDistroData())
```

Comment:
One thing to call out here is that this function only logs the new distro -- it doesn't log both the original and the new and handle things intelligently the way that Project Settings does, and fixing this was initially intended to be a goal of changing to the new UI so we wouldn't have to handle this multiple times (https://jira.mongodb.org/browse/EVG-14885). That being said, I don't see it as an explicit scope goal, but it is still in the project. Is this still something we're going to consider improving? (can still be work outside this ticket but then we may want to add a todo here to log more intelligently)